testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/federation/kerberos/AbstractKerberosSingleRealmTest.java

```diff
@@ -131,35 +132,19 @@ public void usernamePasswordLoginTest() throws Exception {
         // Change editMode to READ_ONLY
         updateProviderEditMode(UserStorageProvider.EditMode.READ_ONLY);
 
-        // Login with username/password from kerberos
-        changePasswordPage.open();
-        loginPage.assertCurrent();
-        loginPage.login("jduke", "theduke");
-        changePasswordPage.assertCurrent();
-
-        // Bad existing password
-        changePasswordPage.changePassword("theduke-invalid", "newPass", "newPass");
-        Assert.assertTrue(driver.getPageSource().contains("Invalid existing password."));
-
         // Change password is not possible as editMode is READ_ONLY
-        changePasswordPage.changePassword("theduke", "newPass", "newPass");
-        Assert.assertTrue(
-                driver.getPageSource().contains("You can't update your password as your account is read-only"));
+        Assert.assertFalse(AccountHelper.updatePassword(testRealmResource(), "theduke", "newPass"));
 
         // Change editMode to UNSYNCED
         updateProviderEditMode(UserStorageProvider.EditMode.UNSYNCED);
 
         // Successfully change password now
-        changePasswordPage.changePassword("theduke", "newPass", "newPass");
-        Assert.assertTrue(driver.getPageSource().contains("Your password has been updated."));
-        changePasswordPage.logout();
+        Assert.assertTrue(AccountHelper.updatePassword(testRealmResource(), "theduke", "newPass"));
 
         // Login with old password doesn't work, but with new password works
         loginPage.login("jduke", "theduke");
         loginPage.assertCurrent();
         loginPage.login("jduke", "newPass");
-        changePasswordPage.assertCurrent();
```

Comment:
Would be good to verify the login was really successful? In the following steps, there's only used `spnegoLogin()`, which only sends HTTP requests.